src/plugins/minimal_scene/EngineToQtInterface.cc

```diff
@@ -0,0 +1,176 @@
+{
+  /// \brief Private data class for EngineToQtInterfacePrivate
+  class EngineToQtInterfacePrivate
+  {
+    /// \brief FBO texture. Stores uploaded GPU -> CPU.
+    /// Used only during fallback.
+    public: GLuint fallbackTexture = 0;
+
+    /// \brief Stores downloaded GPU -> CPU. Used only during fallback
+    public: gz::rendering::ImagePtr fallbackImage;
+
+    /// \brief Qt's OpenGL context. Used only during fallback.
+    /// We need it to render using Qt's context.
+    public: QOpenGLContext *glContext = nullptr;
+  };
+}
+}
+}
+// clang-format on
+
+using namespace gz;
+using namespace gui;
+using namespace plugins;
+
+/////////////////////////////////////////////////
+EngineToQtInterface::EngineToQtInterface(QOpenGLContext *_glContext) :
+  dataPtr(std::make_unique<EngineToQtInterfacePrivate>())
+{
+  this->dataPtr->glContext = _glContext;
+}
+
+/////////////////////////////////////////////////
+EngineToQtInterface::~EngineToQtInterface()
+{
+  this->DestroyFallbackTexture();
+}
+
+/////////////////////////////////////////////////
+void EngineToQtInterface::CreateFallbackTexture()
+{
+  static bool bWarnedOnce = false;
+
+  if (!bWarnedOnce)
+  {
+    gzwarn
+      << "Using fallback method to render to Qt. Things will be SLOW.\n"
+         "Try another API (e.g. OpenGL vs Vulkan) or build against a newer "
+         "Qt version\n";
+    bWarnedOnce = true;
+  }
+
+  this->DestroyFallbackTexture();
+
+  GZ_ASSERT(this->dataPtr->fallbackTexture == 0, "Invalid State!");
+
+  QOpenGLFunctions *glFuncs = this->dataPtr->glContext->functions();
+
+  glFuncs->glGenTextures(1u, &this->dataPtr->fallbackTexture);
+
+  glFuncs->glBindTexture(GL_TEXTURE_2D, this->dataPtr->fallbackTexture);
+  glFuncs->glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_BASE_LEVEL, 0);
+  glFuncs->glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAX_LEVEL, 0);
+  glFuncs->glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MIN_FILTER, GL_NEAREST);
+  glFuncs->glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAG_FILTER, GL_NEAREST);
+  glFuncs->glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_S, GL_CLAMP_TO_EDGE);
+  glFuncs->glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_T, GL_CLAMP_TO_EDGE);
+  glFuncs->glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_WRAP_R, GL_CLAMP_TO_EDGE);
+  glFuncs->glTexParameteri(GL_TEXTURE_2D, GL_TEXTURE_MAX_LEVEL, 0u);
+
+#ifndef __APPLE__
+  this->dataPtr->glContext->extraFunctions()->glTexStorage2D(
+    GL_TEXTURE_2D, 1u, GL_RGBA8, GLsizei(this->dataPtr->fallbackImage->Width()),
+    GLsizei(this->dataPtr->fallbackImage->Height()));
+#else
+  glFuncs->glTexImage2D(GL_TEXTURE_2D, 0, GL_RGBA8,
+                        GLsizei(this->dataPtr->fallbackImage->Width()),
+                        GLsizei(this->dataPtr->fallbackImage->Height()), 0,
+                        GL_RGBA, GL_BYTE, nullptr);
+#endif
+}
+
+/////////////////////////////////////////////////
+void EngineToQtInterface::DestroyFallbackTexture()
+{
+  glDeleteTextures(1, &this->dataPtr->fallbackTexture);
```

Comment:
windows build error here. Maybe the following helps

```diff
-  glDeleteTextures(1, &this->dataPtr->fallbackTexture);
+  QOpenGLFunctions *glFuncs = this->dataPtr->glContext->functions();
+  glFuncs->glDeleteTextures(1, &this->dataPtr->fallbackTexture);
```

Reply:
Agree! That'd probably be the best solution!